src/main/java/org/prgrms/springbootbasic/controller/api/ApiVoucherController.java

```diff
@@ -0,0 +1,91 @@
+import java.util.UUID;
+import org.prgrms.springbootbasic.controller.VoucherType;
+import org.prgrms.springbootbasic.dto.CreateVoucherRequest;
+import org.prgrms.springbootbasic.dto.VoucherDTO;
+import org.prgrms.springbootbasic.dto.VoucherListResponse;
+import org.prgrms.springbootbasic.service.VoucherService;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.ResponseStatus;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+@RequestMapping("/api/v1/vouchers")
+public class ApiVoucherController {
+
+    private static final DateTimeFormatter formatter = DateTimeFormatter.ofPattern("yyyy-MM-dd");
+    private final VoucherService voucherService;
+
+    public ApiVoucherController(VoucherService voucherService) {
+        this.voucherService = voucherService;
+    }
+
+    @GetMapping
+    @ResponseStatus(HttpStatus.OK)
+    public VoucherListResponse findAllVouchers() {
+        var vouchers = voucherService.findAll();
+        return new VoucherListResponse(toVoucherDTOs(vouchers));
+    }
+
+    @GetMapping("/search/{voucherType}")
+    @ResponseStatus(HttpStatus.OK)
+    public VoucherListResponse findVoucherUsingType(
+        @PathVariable("voucherType") VoucherType voucherType) {
+        return new VoucherListResponse(
+            toVoucherDTOs(voucherService.findVoucherUsingType(voucherType)));
+    }
+
+    @GetMapping("/search")
+    @ResponseStatus(HttpStatus.OK)
+    public VoucherListResponse findVoucherUsingCreatedAt(
+        @RequestParam("start") String start, @RequestParam("end") String end) {
+        LocalDateTime startTime = LocalDate.parse(start, formatter).atStartOfDay();
+        LocalDateTime endTime = LocalDate.parse(end, formatter).atStartOfDay();
+
+        return new VoucherListResponse(
+            toVoucherDTOs(voucherService.findVoucherUsingCreatedAt(startTime, endTime)));
+    }
+
+    @PostMapping
+    public ResponseEntity<Void> createVoucher(
+        @RequestBody CreateVoucherRequest createVoucherRequest) throws URISyntaxException {
+        voucherService.createVoucher(
+            createVoucherRequest.getVoucherType(),
+            createVoucherRequest.getAmount() == null ? 0 : createVoucherRequest.getAmount(),
+            createVoucherRequest.getPercent() == null ? 0 : createVoucherRequest.getPercent());
+        return ResponseEntity.created(new URI("/api/v1/vouchers"))
```

Comment:
멘토님 피드백 덕분에 new URI, URI create()를 사용하는 기준을 알았습니다! 🙇‍♂️

> This method is provided for use in situations where it is known that
     * the given string is a legal URI, for example for URI constants declared
     * within in a program, and so it would be considered a programming error
     * for the string not to parse as such.  The constructors, which throw
     * {@link URISyntaxException} directly, should be used situations where a
     * URI is being constructed from user input or from some other source that
     * may be prone to errors.